core/client/src/main/java/alluxio/client/file/policy/BlockLocationPolicy.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.policy;
+
+import alluxio.annotation.PublicApi;
+import alluxio.client.block.BlockWorkerInfo;
+import alluxio.util.CommonUtils;
+import alluxio.wire.WorkerNetAddress;
+
+/**
+ * <p>
+ * Interface for determine the Alluxio worker location to serve a block write or UFS block read.
```

Comment:
We  currently do not use a location policy to determine where to read an Alluxio block though we could.  The current way to server Alluxio read is similar as LocalFirstPolicy